apps/solution/src/components/EditPerson/UsePerson.tsx

```diff
@@ -0,0 +1,16 @@
+import { useEffect, useState } from 'react';
+import { getPeopleById, PersonModel } from '../../api/person';
+
+export default function usePerson(id: string) {
+  const [people, setPeople] = useState<PersonModel | null>(null);
+
+  useEffect(() => {
+    getPeopleById(id).then(setPeople);
+  }, [id]);
+
+  const refreshPerson = (): void => {
+    getPeopleById(id).then(setPeople);
+  };
```

Comment:
Refresh is kind of an anti pattern and should be used only if necessary. I don't think it's our case here since we should be able (I didn't check) to get the updated data from the update.

Since we can extract the entire logic here, we would have a function that calls the update api, and a update(newPerson).then(updatedPerson:PersonModel => setPeople(updatedPerson).

Which means we don't need to make another call, and we update the state here which will cascade.

What do you think ?

Reply:
I think it's possible since the API responds with the updated person. Also, should I move the helper functions to some kind of Utlis folder? We might need to use `unNormalizeDate` and `nomalizeDate` elsewhere.

Reply:
> I think it's possible since the API responds with the updated person. Also, should I move the helper functions to some kind of Utlis folder? We might need to use `unNormalizeDate` and `nomalizeDate` elsewhere.

I am still unsure about those two functions because I find it strange that we handle the date as a string in our model. For me it would be better to handle everything as `Date` (so no conversions needed anywhere apart from display related things) and modifying the data we send to the API to match its data type. What do you think @JonathanGeoffroy @KOCNS1 ?

Reply:
We are using a simple JsonServer where the date are strings. Strangely enough, It results in an `Invalide date` when I try to convert some of them to a real `Date`. I managed to fix it with Moment but it was too having for nothing in our case. 
Dates in JS I guess 👽🤷‍♂️

Reply:
Americans and their date, it's the english format so `new Date("27/10/2015")` would throw invalid, but new `Date("10/27/2015")` will be correct ...

Reply:
but it works with `Date("02/10/1974")` 🧐

Reply:
Cause it's February the 10th of 1974, it's valid

27/10/2015 is the 10th of the 27th month ... which doesn't exist

Reply:
Now I'm focusing on date format , my guess is that's rather a backend issue, as it doesn't provide an largely-compatible, easy-to-read date format.
What about changing dates inthe JSON file to [ISO-8601](https://en.wikipedia.org/wiki/ISO_8601) ?

Reply:
ISO-8601 is always the right answer

Reply:
Changes made ;)